src/pages/availability-panel.tsx

```diff
@@ -17,6 +18,18 @@ const AvailabilityPanel: FC = () => {
   const [isMemberLoading, setIsMemberLoading] = useState<boolean>(true);
   const [refreshData, setRefreshData] = useState<boolean>(false);
 
+  const getAndSetIdleUsers = async () => {
+    try {
+      const [idleUserNames, error] = await fetchIdleUsers();
+      setIdleMembersList(idleUserNames);
+      setError(error);
+    } catch (err) {
+      setError(true);
+    } finally {
+      setIsMemberLoading(false);
+    }
+  };
+
   useEffect(() => {
     const fetchTasks = async () => {
```

Comment:
We could move the `fetchTasks` function outside `useEffect`, but the reason I haven't done that because I felt like it is not in the scope of this PR.

Reply:
Will moving this function outside increase the scope of this PR, if there is no new item being added?

Reply:
It would not necessarily increase the scope, but taking up such changes in a seperate refator PR would make more sense right, as I can see there are quite a few refactors required?